src/commands/preview/request_payload.rs

```diff
@@ -0,0 +1,55 @@
+use url::Url;
+use uuid::Uuid;
+
+use super::http_method::HTTPMethod;
+
+pub struct RequestPayload {
+    pub method: HTTPMethod,
+    pub https: u8,
+    pub session: String,
+    pub protocol: String,
+    pub domain: String,
+    pub path: String,
+    pub query: String,
+    pub browser_url: String,
+    pub service_url: String,
+    pub body: Option<String>,
+}
+
+impl RequestPayload {
+    pub fn create(method: HTTPMethod, url: Url, body: Option<String>) -> RequestPayload {
+        let session = Uuid::new_v4().to_simple().to_string();
+        let https = if url.scheme() == "https" { 1 } else { 0 };
+
+        let protocol = if https == 1 { "https://" } else { "http://" }.to_string();
+        let domain = url.domain().unwrap().to_string();
+        let path = url.path().to_string();
+        let query = format!("?{}", url.query().unwrap_or(""));
```

Comment:
this line makes the default url send `https://example.com/?` which we don't want - I'd recommend changing this to 

```rust
let query = match url.query() {
  Some(query) => format!("?{}", query),
  None => "".to_string()
}
```

Reply:
Good point -- hadn't noticed this and have added this now!